packages/@aws-cdk/aws-ecs/lib/cluster.ts

```diff
@@ -432,13 +432,28 @@ export class Cluster extends Resource implements ICluster {
         default:
           // Amazon ECS-optimized AMI for Amazon Linux 2
           autoScalingGroup.addUserData(`echo ECS_CLUSTER=${this.clusterName} >> /etc/ecs/ecs.config`);
+          let awsVpcBlockImds = options.awsVpcBlockIMDS === true;
           if (!options.canContainersAccessInstanceRole) {
             // Deny containers access to instance metadata service
             // Source: https://docs.aws.amazon.com/AmazonECS/latest/developerguide/instance_IAM_role.html
             autoScalingGroup.addUserData('sudo iptables --insert FORWARD 1 --in-interface docker+ --destination 169.254.169.254/32 --jump DROP');
             autoScalingGroup.addUserData('sudo service iptables save');
             // The following is only for AwsVpc networking mode, but doesn't hurt for the other modes.
+            awsVpcBlockImds = true;
+          }
+          if (awsVpcBlockImds) {
```

Comment:
This is `false` if `canContainersAccessInstanceRole`  is `true` and `awsVpcBlockIMDS` is `false` or `undefined`. This logic also allows for `awsVpcBlockImds` to be `true` even if `canContainersAccessInstanceRole` is `false`.

The reason this is a little complicated is because I was trying to maintain existing behaviour.

Reply:
Ideally with the addition of `awsVpcBlockIMDS` to `AddAutoScalingGroupCapacityOptions` we can remove the automatic setting of `awsVpcBlockIMDS` to `true` when `canContainersAccessInstanceRole` is `false` but that will change existing deployments.